src/index.tsx

```diff
@@ -310,24 +314,46 @@ class PyflyByWidget extends Widget {
   }
 
   restoreNotebookAfterTidyImports(cellArray: any, imports: any): void {
-    const { cellIndex } = this._findAndSetImportCoordinates();
     const cells = this._context.model.cells;
     for (let i = 0; i < cellArray.length; ++i) {
       const cell = cells.get(i);
       const model = cell.sharedModel;
       model.setSource(cellArray[i].text);
     }
-    const joined_imports = imports.join('\n').trim();
-    if (cells.get(0).sharedModel.getSource().length === 0) {
-      cells.get(0).sharedModel.setSource(joined_imports);
-    } else {
-      this._context.model.sharedModel.insertCell(cellIndex, {
-        source: joined_imports,
+
+    const joinedImports = imports.trim();
+    const { cellIndex } = this._findAndSetImportCoordinates();
+
+    const remainingCodeInPyflybyCell = extractCodeFromPyflybyCell(
+      cells.get(cellIndex)
+    );
+
+    if (remainingCodeInPyflybyCell !== '') {
+      // Remove import statements from the current pyflyby cell
+      cells
```

Comment:
Why set the source twice in a row - first to '' then to remaining code?

Reply:
Left in by accident. Removed.